base/task/common/task_annotator.cc

```diff
@@ -282,9 +358,19 @@ TaskAnnotator::ScopedSetIpcHash::ScopedSetIpcHash(
 TaskAnnotator::ScopedSetIpcHash::ScopedSetIpcHash(
     uint32_t ipc_hash,
     const char* ipc_interface_name)
+#if defined(STARBOARD)
+    :
+#else
     : resetter_(&current_scoped_ipc_hash, this),
+#endif
       ipc_hash_(ipc_hash),
-      ipc_interface_name_(ipc_interface_name) {}
+      ipc_interface_name_(ipc_interface_name) {
+#if defined(STARBOARD)
+  EnsureThreadLocalHashKeyInited();
+  scoped_reset_value_ = SbThreadGetLocalValue(s_thread_local_hash_key);
+  SbThreadSetLocalValue(s_thread_local_hash_key, this);
```

Comment:
This one is a bit strange with setting `this`. Are there guarantees that the pointer would not bet taken out after the object is destroyed?

Reply:
You can see that in the destructor of this class we reset the value to what it was initially, as is done upstream, so `this` would only be stored as long as the class is around